openedx/core/djangoapps/xblock/runtime/runtime.py

```diff
@@ -94,8 +111,74 @@ def local_resource_url(self, block, uri):
         return absolute_url
 
     def publish(self, block, event_type, event_data):
-        # TODO: publish events properly
-        log.info("XBlock %s has published a '%s' event.", block.scope_ids.usage_id, event_type)
+        """ Handle XBlock events like grades and completion """
+        special_handler = self.get_event_handler(event_type)
+        if special_handler:
+            special_handler(block, event_data)
+        else:
+            # Why don't we also log events even when special_handler is called?
+            # Not sure, but this if/else is what the similar code in module_render.py was doing.
+            self.log_event_to_tracking_log(block, event_type, event_data)
```

Comment:
I thought the idea was to allow specific events to fill in their own details while still having a generic fallback?

Reply:
Yeah, I just found it weird because in this file it doesn't look like any log gets emitted for grades, but if you follow the logic through the various signals and handlers, it does end up emitting the log. I've removed this comment while squashing.